crates/ursa-rpc-service/src/server.rs

```diff
@@ -56,9 +60,10 @@ where
             .layer(Extension(self.rpc_server.clone()))
     }
 
-    pub fn http_app(&self, metrics: Option<Router>) -> Router {
+    pub fn http_app(&self, index_provider: Router, metrics: Option<Router>) -> Router {
         Router::new()
             .merge(http::routes::network::init::<S>())
+            .merge(index_provider)
```

Comment:
An improvement here would to merge using a Ismaili syntax as the other like “ http::routes::metrics::” something like that

Reply:
It'd need a bigger refactor where rpc-service would depend on index provider and metrics to do that. Vs just externally passing them in